src/main/kotlin/StringAddCalculator.kt

```diff
@@ -0,0 +1,28 @@
+class StringAddCalculator {
+    fun add(text: String?): Int {
+        text ?: return 0
+        if (text.isEmpty()) {
+            return 0
+        }
+
+        val numbers = splitWithDelimiter(text)
+
+        return numbers.sumOf { toPositiveInt(it) }
+    }
+
+    private fun splitWithDelimiter(text: String): List<String> {
+        if (text.matches(Regex("//(.)\n(.*)"))) {
+            val customDelimiter = text[2]
+            return text.substring(4).split(customDelimiter)
```

Comment:
```suggestion
            return text.substring(4)
                .split(customDelimiter)
```
객체지향 생활 체조 원칙
4. 한 줄에 점을 하나만 사용한다

Reply:
저 원칙이 나온 목적이 메시지를 보내라 라는 맥락에서 나온 것으로 생각이 되는데요
지금의 케이스에서는 포메팅 더 관련이 있지 않을까요? 그래서 포메팅만 문제가 없다고 생각이 없다면 안 해도 괜찮을 것 같은데, 어떻게 생각하시나요?

Reply:
네 해당 규칙은 계속해서 깊게 객체를 관여하면서 사용하지 말라는 의미에서 나온 개념이 맞아요.
개인적으로는 그렇기 때문에 극단적으로 해당 규칙을 지키게 되었을 때 깊어지는 체이닝 구조가 눈에 더 들어오고 리팩터링을 할 수 있는 신호를 준다는 생각입니다.